### ISSUE TITLE
Accessing Db inside RunWithinApplication

### ISSUE THREAD

**Issue by malx122** [Starcounter/Starcounter#3982]
Hi!
Trying to accessing the Db inside a `Starcounter.RunWithinApplication("myinternalapp"...` always ends up with `Unknown class XX` where XX is the class given with FULL namespace.

Example:
```
StarcounterEnvironment.RunWithinApplication("colab_common", () =>
{
  var mdi = Db.SQL<Starcounter.Metadata.Index>("SELECT i FRO Starcounter.Metadata.\"Index\" i WHERE Name = ?", name).First;       
});
```
Results in
```
Starcounter.DbException: ScErrFailingEntrypoint (SCERR4259): The user code entrypoint method raised an exception. Failed to process query: SELECT i FROM Starcounter.Metadata."Index" i WHERE Name = ?: Unknown class Starcounter.Metadata.Index. Unknown property or alias Name. 
End position: -1
```

I'm guessing this has to do with the isolation of apps you are doing but I would really need a way to load the db libraries into my "internal custom app" even if I have to tell starcounter manually to do so. I'm calling `RunWithinApplication` from an app that has full access to the wanted classes.

Believe it or not, but this need actually comes from work of making apps more isolated...

**Comment by dan31** [comment]
CC @alemoi @warpech 
```RunWithinApplication``` is not documented on ```.io``` apart from this one https://starcounter.io/hello-world/mapper-app-images-hello-world-part-8/ which appears to be added by @Mackiovello. What is this API and what is our recommendation to use it? What @malx122 is doing wrong?

**Comment by per-samuelsson** [comment]
> I'm guessing this has to do with the isolation of apps

I'm quite sure it has, yes.

**Comment by warpech** [comment]
I used this API to make the code in https://starcounter.io/hello-world/mapper-app-images-hello-world-part-8/ allow me to create a wrapper handler. This surpasses the limitation of `OntologyMap` being able to map a certain URL only once.

I agree that `RunWithinApplication` this needs to be documented better or changed to be more obvious.

On the other hand, I feel that `RunWithinApplication` is not intended for general use. @malx122 can you elaborate on what's your need to use it?

**Comment by malx122** [comment]
Apart from our apps being separate in function, there is of course common code that are shared between the apps to avoid duplication of code. (Both functional code and database classes). This is done in a shared library and it will of course work independent on how many of our apps are running.

In this, there is also common handlers that are needed for some common configurations and other stuff. This I solved at first by making duplicates of all handlers (done in the shared library so just complex, but not duplicate, code) so for example:
```
"/colab_organizer/configuration"
"/colab_actionboard/configuration"
```
both showing the same thing including both a "common" and a mapped app configuration menu. This since I can't map the common menu in that it would make one copy for each app. This actually worked, but is quite complex and really made in dependent on an internal API.

So by using RunWithinApplication I want the first started colab app to register the "colab_common" app that includes the common uri handlers so I just get a:
```
"/colab_common/configuration"
```
This way I can map menu and so on together in the same way as everything else making it prepared for whatever other change coming to the mapping system and not giving you another "mapping" case you need to solve...

It can't be a separate app since it would violate the starcounter dependency between apps-policy.

**Comment by dan31** [comment]
@alemoi I am sort of fluctuating on what type of problem that is. Please put your eye on this one?

**Comment by alemoi** [comment]
> RunWithinApplication is not documented on .io apart from this one https://starcounter.io/hello-world/mapper-app-images-hello-world-part-8/ which appears to be added by @Mackiovello. What is this API and what is our recommendation to use it? What @malx122 is doing wrong?

I would say that I would make `RunWithinApplication` internal. Simply because it violates the principle of apps not knowing about each other.

> This surpasses the limitation of OntologyMap being able to map a certain URL only once.

@warpech this should be solved with Blending.

**Comment by warpech** [comment]
Could you pls check if https://starcounter.io/hello-world/mapper-app-images-hello-world-part-8/ can be realised with `Blending`, without `RunWithinApplication`?

If yes then perfect for my case, though that does not fix the problem in the OP.

**Comment by andwah** [comment]
> I would say that I would make RunWithinApplication internal.

Then it would not be possible to do a Database Editor App ?
